pages/cnpj-generator.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+  <v-layout column justify-center align-center>
+    <v-snackbar v-model="snackbar" class="cnpj-generator__snack-bar">
+      Cnpj copied to clipboard!
+    </v-snackbar>
+    <h1 class="font-weight-thin mb-5">Cnpj Generator</h1>
+    <div class="my-2">
+      <v-btn class="cnpj-generator__button" @click="generateCnpj()"
+        >Generate</v-btn
+      >
+    </div>
+    <v-flex class="cnpj-generator__result-wrapper">
+      <p v-if="result.cnpj" class="cnpj-generator__result">
+        Result: {{ result.cnpj }}
+        <v-icon
+          class="cnpj-generator__copy-button"
+          @click="saveToClipBoard(result.cnpj)"
+          >mdi-content-copy</v-icon
+        >
+      </p>
+      <p v-if="result.maskCnpj" class="cnpj-generator__result">
+        Result with mask: {{ result.maskCnpj }}
+        <v-icon
+          class="cnpj-generator__copy-button"
+          @click="saveToClipBoard(result.maskCnpj)"
+          >mdi-content-copy</v-icon
+        >
+      </p>
+    </v-flex>
+  </v-layout>
+</template>
+
+<script>
+import Vue from 'vue'
+
+export default Vue.extend({
+  name: 'CnpjGenerator',
+  data: () => ({
+    result: { cnpj: '', maskCnpj: '' },
+    snackbar: false
+  }),
+  computed: {},
+
+  methods: {
+    saveToClipBoard(text) {
+      navigator.clipboard.writeText(text)
+      this.snackbar = true
+    },
+
+    random(n) {
+      return Math.round(Math.random() * n)
+    },
+
+    mod(dividend, divider) {
+      return Math.round(dividend - Math.floor(dividend / divider) * divider)
+    },
+
+    generateCnpj() {
+      const n = 9
+      const n1 = this.random(n)
+      const n2 = this.random(n)
+      const n3 = this.random(n)
+      const n4 = this.random(n)
+      const n5 = this.random(n)
+      const n6 = this.random(n)
+      const n7 = this.random(n)
+      const n8 = this.random(n)
+      const n9 = 0
+      const n10 = 0
+      const n11 = 0
+      const n12 = 1
```

Comment:
I think there's some room for simplification here. What if you do something like the suggested below:

```suggestion
      const [n1, n2, n3, n4, n5, n6, n7, n8, n9, n10, n11, n12] = new Array(8)
        .fill(0)
        .concat([0, 0, 0, 1])
        .map(() => this.random(n))
```

Reply:
Maybe there is room for more improvements from this method, but I'm gonna dig into it later.

Another thing that I missed is: is there any source from where you got this algorithm? It would be nice to insert as a comment in the code to people to see where you get it.

Reply:
Doesn't `.map(() => this.random(n))` change `n9, n10, n11, n12` when they must be `[0, 0, 0, 1]`? Maybe change position between `map` and `concat` should handle it

Reply:
You're right. Inverting suggestion.